contributors/design-proposals/componentconfig-dump.md

```diff
@@ -0,0 +1,164 @@
+```
+
+similar to /metrics endpoints for K8s components and likely to be authenticated in much the same way.
+
+By way of an `HTTP GET` the output would be:
+
+```
+{
+    "kind": "configuration-list",
+    "metadata": {
+        "component": "kubelet",
+        "id": "818B908B-D053-CB11-BC8B-EEA826EBA090",
+        "timeStamp": "2016-09-13T14:47:35+00:00",
+    },
+    "configurations": [
+        {
+            "type": "componentconfigs",
+            "metadata": {
+                "version": "v1",
+                ...
+            },
+            "list": [
+                cloud-provider": "aws",
+                "aGroupName": {
+                    "aKey": "aValue",
+                    ...
+                },
+                ...
+            ]        
+        },
+        {
+            "type": "componentconfigs",
+            "metadata": {
+                "version": "v2",
+                ...
+            },
+            "list": [
+                cloud-provider": "aws",
+                "bGroupName": {
+                    "bKey": "bValue",
+                    ...
+                },
+                ...
+            ]        
+        },
+        {
+            "type": "additionalconfigs",
+            "list": [
+                "bKey": "bValue",
+                ...
+            ]
+        }
+    ]
+}
+```
+
+Those componentconfigs that are to be associated with a particular component will be annotated so as to be labeled for collection and served by the proposed endpoint.
+
+## Alternatives and Further Discussions
+
+Here we mention points of contentions and alternatives as we make progress in the proposal.
+
+* Security/Auth has been brought up as a point that strongly affects the proposed endpoint. Ideas around component endpoints and auth have been discussed for `/metrics` and it has been noted that SIG-Auth may be assigned the `/configs` endpoint to ensure auth is handled correctly.
+* The addition of an api change/addition to APIServer has been discussed, and for reasons of security it was agreed to be left out as a potential addition in the future.
+* The addition of kubectl/kubeadmin interactions has also been scrapped for reasons of security with possible addition in the future, but do not gain us much for our present motivations.
+* It has been agreed that extracting configs from componentconfigs is the way forward.
+
+## Future Work
+
+* Collect all possible configurations that could be extracted. Determine which configurations belong under the compomentconfig object and which do not and have an understanding why.
+
+* In the future configurations will be versioned, so that you can get specific
+version of a given type
+
+    ```
+    /configs/componentconfigs/v1/
+    ```
+
+    where the configurations list will contain only that componentconfig whose version is v1. Note, lack of version specified could result in all versions exposed or only the latest, a choice should be made where the recommendation is for all.
```

Comment:
I would recommend that you just re-use existing machinery and don't invent behavior for the paths here and below.